hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -698,11 +699,12 @@ public void loadAll(Set<K> keys, boolean replaceExistingValues) {
         MapKeySetRequest request = new MapKeySetRequest(name);
         MapKeySet mapKeySet = invoke(request);
         Set<Data> keySetData = mapKeySet.getKeySet();
-        Set<K> keySet = new HashSet<K>(keySetData.size());
+        InflatableSet<K> keySet = new InflatableSet<K>(keySetData.size());
         for (Data data : keySetData) {
             final K key = toObject(data);
```

Comment:
@gurbuzali, @ahmetmircik:

see this: https://github.com/jerrinot/hazelcast/blob/experiments/inflatableSet-lazyEntry/master/hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java#L828-828

No reason to poison the Set with deserilization logic. How does it look? I think we can apply this composition for other methods too.